aws-rds-dbinstance/src/main/java/software/amazon/rds/dbinstance/CreateHandler.java

```diff
@@ -169,12 +180,42 @@ ApiVersion.DEFAULT, safeAddTags(this::createDbInstance)
                             logger
                     );
                 });
-    }
+    }final
 
     private HandlerMethod<ResourceModel, CallbackContext> safeAddTags(final HandlerMethod<ResourceModel, CallbackContext> handlerMethod) {
         return (proxy, rdsProxyClient, progress, tagSet) -> progress.then(p -> Tagging.safeCreate(proxy, rdsProxyClient, handlerMethod, progress, tagSet));
     }
 
+    private String fetchEngine(final ProxyClient<RdsClient> client, final ResourceModel model) {
```

Comment:
I understand why this is needed but it would be great to start seeing our handler logic simplify in the long run by enforcing better API standards on the RDS side.

Reply:
Agreed. At this point, however, we need this method in order to make sure all engines use appropriate workflows